01/hello_test.go

```diff
@@ -1,13 +1,12 @@
-package hello_test
+package hello
```

Comment:
In some ways it's easier to write test code 'from inside'—putting your tests in the same package as the code being tested. However, there is a strong argument that it's better to write tests from outside. One of the main benefits is that you're the first user of your own package: you're importing it and using your own API. If the import path or the package name don't make sense, or the names of your functions read awkwardly when combined with the package, you'll discover that before anyone else does.

Another benefit is that outside-in tests force you to design things in such a way that they can be easily tested. For example, if your function makes HTTP calls, you'll need to make it possible for users of your package to inject their own HTTP client, and to change the URL that you call. Both of these are required for tests, but they're also helpful for users.

There are some cases where it's impossible to test a certain piece of code from outside, or making it possible to do so unnecessarily complicates your API. In those cases we would use 'internal' tests, but we would still put as many tests as possible in an external package.